plugin/goto.py

```diff
@@ -30,71 +54,26 @@ def run(self, edit: sublime.Edit, event: Optional[dict] = None) -> None:
                 debug("unrecognized goto kind:", self.goto_kind)
                 return
             request = request_type(document_position)
-            client.send_request(
-                request, lambda response: self.handle_response(response, pos))
-
-    def handle_response(self, response: Optional[Any], position: int) -> None:
-        def process_response_list(responses: list) -> List[Tuple[str, str, Tuple[int, int]]]:
-            return [process_response(x) for x in responses]
-
-        def process_response(response: dict) -> Tuple[str, str, Tuple[int, int]]:
-            if "targetUri" in response:
-                # TODO: Do something clever with originSelectionRange and targetRange.
-                file_path = uri_to_filename(response["targetUri"])
-                start = Point.from_lsp(response["targetSelectionRange"]["start"])
-            else:
-                file_path = uri_to_filename(response["uri"])
-                start = Point.from_lsp(response["range"]["start"])
-            row = start.row + 1
-            col = start.col + 1
-            file_path_and_row_col = "{}:{}:{}".format(file_path, row, col)
-            return file_path, file_path_and_row_col, (row, col)
-
-        def open_location(window: sublime.Window, location: Tuple[str, str, Tuple[int, int]]) -> None:
-            fname, file_path_and_row_col, rowcol = location
-            row, col = rowcol
-            debug("opening location", file_path_and_row_col)
-            window.open_file(
-                file_path_and_row_col,
-                sublime.ENCODED_POSITION)
-
-        def select_entry(
-                window: sublime.Window,
-                locations: List[Tuple[str, str, Tuple[int, int]]],
-                idx: int,
-                orig_view: sublime.View) -> None:
-            if idx >= 0:
-                open_location(window, locations[idx])
-            else:
-                if orig_view:
-                    window.focus_view(orig_view)
-
-        def highlight_entry(
```

Comment:
These three closures actually did not capture any variables outside the function, so might as well make them plain functions.